packages/wallet/backend/src/paymentPointer/service.ts

```diff
@@ -132,4 +134,84 @@ export class PaymentPointerService implements IPaymentPointerService {
       active: false
     })
   }
+
+  async generateKeyPair(
+    userId: string,
+    paymentPointerId: string
+  ): Promise<{ privateKey: KeyObject; publicKey: KeyObject }> {
+    const paymentPointer = await PaymentPointer.query().findById(
+      paymentPointerId
+    )
+
+    if (!paymentPointer) {
+      throw new NotFound()
+    }
+
+    // Check if the user owns the payment pointer.
+    // This function throws a NotFoundException.
+    await this.deps.accountService.findAccountById(
+      paymentPointer.accountId,
+      userId
+    )
+
+    // Generate the key pair
+    const { privateKey } = generateKeyPairSync('ed25519')
+    const publicKey = createPublicKey(privateKey)
+
+    await PaymentPointer.query()
+      .findById(paymentPointerId)
+      .patch({
+        publicKey: JSON.stringify(publicKey)
+      })
+
+    return { privateKey, publicKey }
+  }
+
+  async registerKey(userId: string, paymentPointerId: string): Promise<void> {
+    const paymentPointer = await PaymentPointer.query().findById(
+      paymentPointerId
+    )
+
+    if (!paymentPointer) {
+      throw new NotFound()
+    }
+
+    // Check if the user owns the payment pointer.
+    // This function throws a NotFoundException.
+    await this.deps.accountService.findAccountById(
```

Comment:
same as above, define a method in payment pointer service to use it in both places